Master/healthcheck.js

```diff
@@ -0,0 +1,111 @@
+/* ----------------------------------------------------------------------------
+ * @copyright (C) 2019, Alert Logic, Inc
+ * @doc
+ * 
+ * Various Event hub collector health checks.
+ * The last error code is EHUB000006
+ * 
+ * @end
+ * ----------------------------------------------------------------------------
+ */
+ 
+const async = require('async');
+const azure = require('azure');
+const parse = require('parse-key-value');
+
+function initArmEhub(master) {
+    const azureCreds = master.getApplicationTokenCredentials();
+    const subscriptionId = master.getConfigAttrs().subscription_id;
+    return azure.createEventHubManagementClient(azureCreds, subscriptionId);
+}
+
+function formatSdkError(master, alErrorCode, err) {
+    if (typeof err === 'string' || err instanceof String) {
+        return master.errorStatusFmt(alErrorCode, err);
+    } else if (typeof err === 'object') {
+        return master.errorStatusFmt(alErrorCode, JSON.stringify(err));
+    }
+}
+
+function getEhubNsName() {
+    const connectionParams = parse(process.env.APP_LOG_EHUB_CONNECTION);
+    return connectionParams.Endpoint.split(/[\.\/]/g)[2];
+}
+
+function checkEventHubNamespace(master, ns, callback) {
+    const pState = ns.provisioningState;
+    //const status = ns.status;
+    if (pState === 'Succeeded') {
+        return callback(null, ns);
+    } else {
+        const err = master.errorStatusFmt(
+            'EHUB000001',
+            `Event Hub Namespace state is not ok. Namespace = ${ns.name}, provisioningState = ${pState}`);
+        
+        return callback(err);
+    }
+}
+
+function checkEventHub(master, eventHubs, callback) {
+    const check = eventHubs.reduce(function(acc, ehub) {
+        acc.logHubExists = acc.logHubExists || ehub.name === 'alertlogic-log';
+        if (acc.error) {
+            return acc;
```

Comment:
IIUC, this will return only the first error; if both event hubs are in error, we won't get the 2nd error, right?

Reply:
right. This is intended.